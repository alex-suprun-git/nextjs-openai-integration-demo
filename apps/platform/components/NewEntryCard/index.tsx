'use client';

import { useTranslations } from 'next-intl';
import { useRouter } from 'next/navigation';

export const NewEntryCard = () => {
  const router = useRouter();

  const t = useTranslations('JournalList');

  const handleOnClick = async () => {
    router.push('/new-entry');
  };

  return (
    <div
      onClick={handleOnClick}
      className="card h-[128px] cursor-pointer bg-yellow-200 text-primary-content"
    >
      <div className="card-body">
        <span data-testid="new-entry-button" className="min-h-20 text-2xl font-bold text-gray-900">
          {t('buttons.newEntry')}
        </span>
      </div>
    </div>
  );
};

export default NewEntryCard;
