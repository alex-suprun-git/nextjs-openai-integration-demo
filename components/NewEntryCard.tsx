'use client';

import { createNewEntry } from '@/utils/api';
import { useRouter } from 'next/navigation';
import { FC } from 'react';

const NewEntryCard: FC<{}> = () => {
  const router = useRouter();

  const handleOnClick = async () => {
    const entryData = await createNewEntry();
    router.push(`/journal/${entryData.id}`);
  };

  return (
    <div className="cursor-pointer overflow-hidden rounded-lg bg-white shadow">
      <div className="px-4 py-5 sm:p-6" onClick={handleOnClick}>
        <span className="text-3xl">New Entry</span>
      </div>
    </div>
  );
};
export default NewEntryCard;
