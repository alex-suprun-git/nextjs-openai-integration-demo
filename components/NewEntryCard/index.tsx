'use client';

import { useRouter } from 'next/navigation';

const NewEntryCard = () => {
  const router = useRouter();

  const handleOnClick = async () => {
    router.push('/journal/new-entry');
  };

  return (
    <div onClick={handleOnClick} className="card cursor-pointer bg-yellow-200 text-primary-content">
      <div className="card-body">
        <span className="text-2xl font-bold text-gray-900">Write a new memo</span>
      </div>
    </div>
  );
};
export default NewEntryCard;
