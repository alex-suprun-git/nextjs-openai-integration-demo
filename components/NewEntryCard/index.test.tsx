import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi, Mock } from 'vitest';
import { useRouter } from 'next/navigation';
import NewEntryCard from '.';

vi.mock('next/navigation', () => ({
  useRouter: vi.fn(),
}));

describe('NewEntryCard', () => {
  const mockPush = vi.fn();

  beforeEach(() => {
    (useRouter as Mock).mockReturnValue({
      push: mockPush,
    });
  });

  it('renders correctly and matches snapshot', () => {
    const { container } = render(<NewEntryCard />);
    expect(container).toMatchSnapshot();
  });

  it('renders with correct text', () => {
    render(<NewEntryCard />);
    expect(screen.getByText('Write a new memo')).toBeInTheDocument();
  });

  it('navigates to the correct path on click', () => {
    render(<NewEntryCard />);
    fireEvent.click(screen.getByText('Write a new memo'));
    expect(mockPush).toHaveBeenCalledWith('/journal/new-entry');
  });
});
