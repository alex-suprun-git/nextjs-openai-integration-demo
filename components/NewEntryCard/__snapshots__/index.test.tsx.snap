// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewEntryCard > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="card cursor-pointer bg-yellow-200 text-primary-content"
  >
    <div
      class="card-body"
    >
      <span
        class="text-2xl font-bold text-gray-900"
      >
        Write a new memo
      </span>
    </div>
  </div>
</div>
`;
